### PR TITLE
Add initial support for ArubaOS MM product

### PR DESCRIPTION
**Pull request by ykoehler**
More will follow

### CODE REVIEW

**Review by LukasMcClelland — COMMENTED** [review]
Thank you for contributing! 

Does "More to follow" mean that you'll be adding more to this PR? It would great to see an arubaos mock device and some tests as well, if possible

**Comment by ykoehler** [comment]
By "more to follow", I meant that I will likely add more contributions soon to add support for other Aruba products as well, didn't explicitly mean this PR but for sure whatever you guys determine is required for this to get approved I will see if I can add.

Note that I do work for HPE Aruba, but this is a personal project of mine. 

Note that I would be more than happy to add "mock" or "tests" yet I have not really found the documentation/instructions on how to do so, please point me to the correct location and I will be happy to see what I can do about this.

**Comment by ykoehler** [comment]
ok, was able to figure out the mock/unit test part, not sure I got it all right but was able to get a success on my side.

File: src/unicon/plugins/arubaos/__init__.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        super().__init__()
+        self.enable = svc.Enable
+        self.disable = svc.Disable
+        self.config = svc.Config
+        self.rommon = svc.Rommon
+        self.shell = svc.Shell
+        self.switchto = svc.Switchto
+
+
+class ArubaosSingleRPConnection(IosvSingleRpConnection):
```
[the remark this answers is not in the document]
Comment (ykoehler) [reply src/unicon/plugins/arubaos/__init__.py:26]: Not to my knowledge, I will have a look at the generic and see how that works.

**Comment by ykoehler** [comment]
Sorry, not the proper place to ask, but I am kind of still trying to figure out where and how would pyATS goes for a firmware update, any place where I could find examples for other devices or some slack channel, or anything else where I could ask those questions and get support?

**Comment by LukasMcClelland** [comment]
> Sorry, not the proper place to ask, but I am kind of still trying to figure out where and how would pyATS goes for a firmware update, any place where I could find examples for other devices or some slack channel, or anything else where I could ask those questions and get support?

For users who are not Cisco employees, the 2 best channels are our public Webex room and our email support mailer list:

1. https://eurl.io/#r18UzrQVr
2. pyats-support-ext@cisco.com

I would recommend using the Webex room first as there are over 1000 other people in there asking questions and there's tons of good info in there

File: src/unicon/plugins/arubaos/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (LukasMcClelland) [reply src/unicon/plugins/arubaos/__init__.py:26]: Sounds good, if both work well, then I would recommend inheriting from the generic one as it means one less link in the inheritance chain

**Comment by ademz** [comment]
@ykoehler Thank you for your contribution. 

I'm closing this one.

**Pull request closed without merging**